dotnet/src/SemanticKernel/Planning/Plan.cs

```diff
@@ -256,7 +256,10 @@ public async Task<Plan> InvokeNextStepAsync(SKContext context, CancellationToken
 
             var result = await step.InvokeAsync(functionContext, cancellationToken: cancellationToken).ConfigureAwait(false);
```

Comment:
It feels like this should also maintain a list of all function results similar to kernel.runasync. Thoughts?

Reply:
This is one of the problems I see with `Plan` as implementation of `ISKFunction`. Because from the first sight it looks like it's a function that just contains sub-functions, but in some cases, we can see that it's a little bit different. In my opinion - `Function`, as entity, should return single result model (inside result we can have everything), but `Plan`, same as Kernel, should return final result (from last function in plan), but also it can return all function results for better observability and maintenance.

Reply:
```
functionContext.Kernel.RunAsync(step) ...

Reply:
@RogerBarreto In this case there will be still only one result, because you can get multiple results from kernel only if you call multiple functions in pipeline - `kernel.RunAsync("input", func1, func2, func3)`.

Reply:
Actually a Plan Execution will return multiple results.

Check latest @shawncal PR #2914 and see if this resonates.

Reply:
Should a semantic function not also return all the various results from the function the template engine might call?

Reply:
As result `Value`, it will always contain result from itself. While `FunctionResults` may return results from each child function, and I think it should be useful (information like amount of tokens used, model response etc.)